build/ableplayer.dist.js

```diff
@@ -38,14 +43,15 @@
 var AblePlayerInstances = [];
 
 (function ($) {
-	$(document).ready(function () {
-
-		$('video, audio').each(function (index, element) {
-			if ($(element).data('able-player') !== undefined) {
-				AblePlayerInstances.push(new AblePlayer($(this),$(element)));
-			}
-		});
-	});
+	// Initialise all AblePlayerInstances - this can be called
+	// by the React component that imports ableplayer
+  window.getAllAblePlayers = function() {
+    $("video, audio").each(function(index, element) {
+      if ($(element).data("able-player") !== undefined) {
+        AblePlayerInstances.push(new AblePlayer($(this), $(element)));
+      }
+    });
+  };
```

Comment:
The above function shouldn't get triggered at loading time because there's a race-condition where the React component hasn't finished initialising and this runs before it, causing AblePlayer injection code to not trigger properly.

Reply:
I'll rename it to something more indicative of what it's actually doing - `createAblePlayerInstances`